env/env_posix.cc

```diff
@@ -778,6 +779,18 @@ class PosixEnv : public Env {
     return Status::OK();
   }
 
+  virtual Status CheckCompactionSize(const char *path_in_filesystem,
```

Comment:
I don't really feel it fits that well with the Env abstraction. Would it make sense to have a `GetEstimatedFreeSize` function here, and put the logic for checking whether compaction can run closer to the compaction code?

Reply:
Gone now